tests/delete_regress/tests.py

```diff
@@ -379,14 +379,22 @@ def test_self_reference_with_through_m2m_at_second_level(self):
         child = Child.objects.create(name="Juan")
         Book.objects.create(pagecount=500, owner=child)
         PlayedWith.objects.create(child=child, toy=toy, date=datetime.date.today())
-        Book.objects.filter(
-            Exists(
-                Book.objects.filter(
-                    pk=OuterRef("pk"),
-                    owner__toys=toy.pk,
-                ),
-            )
-        ).delete()
+        with self.assertNumQueries(1) as ctx:
+            Book.objects.filter(
+                Exists(
+                    Book.objects.filter(
+                        pk=OuterRef("pk"),
+                        owner__toys=toy.pk,
+                    ),
+                )
+            ).delete()
+
+            sql = ctx.captured_queries[0]["sql"].lower()
+            print(sql)
+
+            # reducing additional select statement if flag is true, when using subquery
+            if connection.features.delete_can_self_reference_subquery:
+                self.assertEqual(sql.count("select"), 1)
         self.assertIs(Book.objects.exists(), False)
```

Comment:
You can drop the `print` call and move this logic outside of the managed context. We also usually put complementary assertions at the end of a test. Since the code is self documenting I also think the comment can be dropped without.

```suggestion
        self.assertIs(Book.objects.exists(), False)
        sql = ctx.captured_queries[0]["sql"].lower()
        if connection.features.delete_can_self_reference_subquery:
            self.assertEqual(sql.count("select"), 1)
```

Reply:
@charettes  thanks for the review and really apologize for that naive `print` call  8 )

Reply:
did the changes right away